src/stream/stream/mod.rs

```diff
@@ -373,6 +375,40 @@ extension_trait! {
             Chain::new(self, other)
         }
 
+            #[doc = r#"
+            Creates an stream which copies all of its elements.
+
+            # Examples
+
+            Basic usage:
+
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use async_std::prelude::*;
+            use std::collections::VecDeque;
+
+            let v: VecDeque<_> = vec![&1, &2, &3].into_iter().collect();
+
+            let mut v_cloned = v.cloned();
+
+            assert_eq!(v_cloned.next().await, Some(1));
+            assert_eq!(v_cloned.next().await, Some(2));
+            assert_eq!(v_cloned.next().await, Some(3));
+            assert_eq!(v_cloned.next().await, None);
+
+            #
+            # }) }
+            ```
+        "#]
+        fn cloned<'a,T>(self) -> Cloned<Self>
```

Comment:
I tryed running this code.
```rust
fn main() -> std::io::Result<()> {
    async_std::task::block_on(async {
        use async_std::prelude::*;
        use std::collections::VecDeque;
        let v: VecDeque<_> = vec![&1, &2, &3].into_iter().collect();
        let v_cloned = v.cloned();
        dbg!(v);
        dbg!(v_cloned);
    });

    Ok(())
}
```


I god this error.
```console
error[E0382]: use of moved value: `v`
 --> src/main.rs:7:14
  |
5 |         let v: VecDeque<_> = vec![&1, &2, &3].into_iter().collect();
  |             - move occurs because `v` has type `std::collections::VecDeque<&i32>`, which does not implement the `Copy` trait
6 |         let v_cloned = v.cloned();
  |                        - value moved here
7 |         dbg!(v);
  |              ^ value used here after move
```

Is it appropriate to move ownership?

Reply:
> Is it appropriate to move ownership?

This should be fine because `cloned` clones each element of the stream, not the stream itself.

Refs: [std::iter::Iterator::cloned](https://doc.rust-lang.org/nightly/core/iter/trait.Iterator.html#method.cloned)

Reply:
thanks!